src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -499,40 +499,79 @@ public static void close(Closeable closeable) {
      * @throws CheckstyleException on failure
      */
     public static URI getUriByFilename(String filename) throws CheckstyleException {
-        // figure out if this is a File or a URL
+        URI uri = getWebOrFilePrefixUri(filename);
+
+        if (uri == null) {
+            uri = getClasspathOrFilepathUri(filename);
+        }
+
+        return uri;
+    }
+
+    /**
+     * Resolves the specified local filename to a URI.
+     *
+     * @param filename name of the file
+     * @return resolved header file URI
+     * @throws CheckstyleException on failure
+     */
+    private static URI getClasspathOrFilepathUri(String filename) throws CheckstyleException {
+        final URI uri;
+        if (new File(filename).exists()) {
+            final File file = new File(filename);
+            uri = file.toURI();
+        }
+        else {
+            uri = getResources(filename);
+        }
+        return uri;
+    }
+
+    /**
+     * Resolves the specified filename containing 'http', 'https', 'ftp',
+     * and 'file' protocols (or any RFC 2396 compliant URL) to a URI.
+     *
+     * @param filename name of the file
+     * @return resolved header file URI or null if URL is malformed or non-existent
+     */
+    public static URI getWebOrFilePrefixUri(String filename) {
         URI uri;
         try {
             final URL url = new URL(filename);
             uri = url.toURI();
         }
-        catch (final URISyntaxException | MalformedURLException ignored) {
+        catch (final URISyntaxException | MalformedURLException ex) {
             uri = null;
         }
+        return uri;
+    }
 
-        if (uri == null) {
-            final File file = new File(filename);
-            if (file.exists()) {
-                uri = file.toURI();
-            }
-            else {
-                // check to see if the file is in the classpath
-                try {
-                    final URL configUrl;
-                    if (filename.charAt(0) == '/') {
-                        configUrl = CommonUtil.class.getResource(filename);
-                    }
-                    else {
-                        configUrl = ClassLoader.getSystemResource(filename);
-                    }
-                    if (configUrl == null) {
-                        throw new CheckstyleException(UNABLE_TO_FIND_EXCEPTION_PREFIX + filename);
-                    }
-                    uri = configUrl.toURI();
-                }
-                catch (final URISyntaxException ex) {
-                    throw new CheckstyleException(UNABLE_TO_FIND_EXCEPTION_PREFIX + filename, ex);
-                }
-            }
+    /**
+     * Gets resources from classpath.
+     *
+     * @param filename name of file
+     * @return URI of file in classpath
+     * @throws CheckstyleException on failure
+     */
+    public static URI getResources(String filename) throws CheckstyleException {
```

Comment:
`getResource`

Reply:
This will not work, since `CommonUtil.class.getResource` is already used within this class.  Renamed to `getResourceFromClassPath()`.